app/gui/src/model/project/synchronized.rs

```diff
@@ -707,7 +707,9 @@ impl model::project::API for Project {
     ) -> BoxFuture<FallibleResult<model::ExecutionContext>> {
         async move {
             let ls_rpc = self.language_server_rpc.clone_ref();
-            let context = execution_context::Synchronized::create(ls_rpc, root_definition);
+            let context_id = Uuid::new_v4();
```

Comment:
Random UUID is created on the IDE side and send to the LS+engine to use it. That's a change to current behavior when no UUID is send the the LS+engine _thinks one up_. As a result the _randomness_ is on the IDE side and the engine behaves (more) deterministically when fed with the same stream of messages.

Reply:
I think this is slightly related to https://github.com/enso-org/enso/issues/6718